tam/grpc-multi-tenant/internal/tenant/ports/grpc.go

```diff
@@ -0,0 +1,125 @@
+package ports
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/tuannguyenandpadcojp/go-training/tam/grpc-multi-tenant/gen/go/tenant/v1"
+	"github.com/tuannguyenandpadcojp/go-training/tam/grpc-multi-tenant/internal/pkg"
+	"github.com/tuannguyenandpadcojp/go-training/tam/grpc-multi-tenant/internal/tenant/app"
+	user_app "github.com/tuannguyenandpadcojp/go-training/tam/grpc-multi-tenant/internal/user/app"
+)
+
+type GrpcServer struct {
+	pb.UnimplementedTenantServiceServer
+	app     app.Application
+	userApp user_app.Application
+}
+
+func NewGrpcServer(app app.Application, userApp user_app.Application) pb.TenantServiceServer {
+	return &GrpcServer{app: app, userApp: userApp}
+}
+
+func (s *GrpcServer) CreateTenant(ctx context.Context, req *pb.CreateTenantRequest) (*pb.CreateTenantResponse, error) {
+	if err := s.validateCreateTenantRequest(req); err != nil {
+		return nil, err
+	}
+
+	tenant, err := s.app.CreateTenant(ctx, app.CreateTenantInput{
+		Name:       req.Name,
+		OwnerEmail: req.OwnerEmail,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.CreateTenantResponse{
+		Id:         tenant.ID,
+		Name:       tenant.Name,
+		OwnerEmail: tenant.OwnerEmail,
+		CreatedAt:  tenant.CreatedAt.Format("2006-01-02T15:04:05Z"),
+	}, nil
+}
+
+func (s *GrpcServer) validateCreateTenantRequest(req *pb.CreateTenantRequest) error {
+	if req.Name == "" || req.OwnerEmail == "" {
+		return fmt.Errorf("name and owner_email are required")
+	}
+	if !pkg.EmailRegex.MatchString(req.OwnerEmail) {
```

Comment:
I prefer expose validation function than pkg variable
```suggestion
	if !pkg.IsValidEmail(req.OwnerEmail) {
```